lib/spack/spack/reporters/cdash.py

```diff
@@ -136,12 +144,15 @@ def build_report(self, filename, report_data):
             build_pos = phases_encountered.index("build")
             phases_encountered.insert(0, phases_encountered.pop(build_pos))
 
-        self.starttime = self.endtime - total_duration
+        self.starttime = self.endtime - duration
         for phase in phases_encountered:
             report_data[phase]['starttime'] = self.starttime
             report_data[phase]['log'] = \
                 '\n'.join(report_data[phase]['loglines'])
             errors, warnings = parse_log_events(report_data[phase]['loglines'])
+            # Cap the number of errors and warnings at 50 each.
+            errors = errors[0:49]
+            warnings = warnings[0:49]
```

Comment:
This is good, hopefully it will get us around the issue we last saw with the reporter hanging forever when there were lots of warnings.